common/djangoapps/student/helpers.py

```diff
@@ -198,6 +220,75 @@ def auth_pipeline_urls(auth_entry, redirect_url=None):
     }
 
 
+def _update_email_opt_in(request, org):
+    """Helper function used to hit the profile API if email opt-in is enabled."""
+
+    email_opt_in = request.POST.get('email_opt_in')
+    if email_opt_in is not None:
+        email_opt_in_boolean = email_opt_in == 'true'
+        preferences_api.update_email_opt_in(request.user, org, email_opt_in_boolean)
+
+
+def enroll(user, course_id, request, check_access):
```

Comment:
@OmarIthawi Actually if you look at the latest edX code, enrollment is still done this way, this is only related to LMS when a user chooses to enroll/unenroll in a course. I do not see how I can use the enrollment/api in this specific case.